agent/lib_monolog.c

```diff
@@ -0,0 +1,416 @@
+  int retval = 0;
+  zval* api = NULL;
+  zend_class_entry* ce = NULL;
+
+  if (0 == nr_php_is_zval_valid_object(logger)) {
+    nrl_verbosedebug(NRL_INSTRUMENT, "%s: Logger object is invalid", __func__);
+    return 0;
+  }
+
+  ce = Z_OBJCE_P(logger);
+  if (NULL == ce) {
+    nrl_verbosedebug(NRL_INSTRUMENT, "%s: Logger has NULL class entry",
+                     __func__);
+    return 0;
+  }
+
+  api = nr_php_get_class_constant(ce, "API");
+  if (NULL == api) {
+    nrl_verbosedebug(NRL_INSTRUMENT, "%s: Logger does not have API", __func__);
+    return 0;
+  }
+
+  if (nr_php_is_zval_valid_integer(api)) {
+    retval = Z_LVAL_P(api);
+  } else {
+    nrl_verbosedebug(NRL_INSTRUMENT,
+                     "%s: expected API be a valid int, got type %d", __func__,
+                     Z_TYPE_P(api));
+  }
+
+  nr_php_zval_free(&api);
+  return retval;
+}
+
+/*
+ * Purpose : Convert $level argument of Monolog\Logger::addRecord to a string
+ * representation of Monolog's log level.
+ *
+ * Params  : Logger instance and Monolog\Logger::addRecord argument list
+ *
+ * Returns : A new string with Monolog's log level name; caller must free
+ */
+static char* nr_monolog_get_level_name(zval* logger,
+                                       NR_EXECUTE_PROTO TSRMLS_DC) {
+  zval* level_name = NULL;
+  char* level_name_string = NULL;
+
+  if (!nr_php_object_has_method(logger, "getLevelName" TSRMLS_CC)) {
+    nrl_verbosedebug(NRL_INSTRUMENT,
+                     "%s: Logger does not have getLevelName method", __func__);
+  } else {
+    zval* level = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+    if (NULL == level) {
+      nrl_verbosedebug(NRL_INSTRUMENT,
+                       "%s: $level not defined, unable to get log level name",
+                       __func__);
+    } else {
+      level_name = nr_php_call(logger, "getLevelName", level);
+    }
+    nr_php_arg_release(&level);
+    if (NULL == level_name) {
+      nrl_verbosedebug(NRL_INSTRUMENT, "%s: expected level_name be valid",
+                       __func__);
+    } else if (!nr_php_is_zval_valid_string(level_name)) {
+      nrl_verbosedebug(NRL_INSTRUMENT,
+                       "%s: expected level_name be a valid string, got type %d",
+                       __func__, Z_TYPE_P(level_name));
+    } else {
+      level_name_string = nr_strdup(Z_STRVAL_P(level_name));
+    }
+    nr_php_zval_free(&level_name);
+  }
+
+  return level_name_string != NULL ? level_name_string : nr_strdup("UNKNOWN");
+}
+
+/*
+ * Purpose : Convert $message argument of Monolog\Logger::addRecord to a string
+ *
+ * Params  : Monolog\Logger::addRecord argument list
+ *
+ * Returns : A new string with Monolog's log message; caller must free
+ */
+
+static char* nr_monolog_get_message(NR_EXECUTE_PROTO TSRMLS_DC) {
+  char* message = NULL;
+  zval* message_arg = NULL;
+
+  message_arg = nr_php_arg_get(2, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (message_arg) {
```

Comment:
To be consistent with other checks in this file for the return value of `nr_php_arg_get()`.

```suggestion
  if (NULL == message_arg) {
```

Reply:
Of course! 🤦